### PR TITLE
Use Inventories instead of ItemStacks where applicable

### PR DESCRIPTION
**Pull request by Aaron1011**
Many parts of the API don't make use of the Inventory API. Instead, they use the old approach of getting and setting `ItemStack`s directly.

This PR modifies several interfaces to return an `Inventory` instead of an `ItemStack`. The necessary setter is removed.

It also removes `EntityEquipEvent`, which is made unnecessary by `EntityEquipmentChangeEvent`.

### CODE REVIEW

**Comment by ST-DDT** [comment]
I like the idea but especially in some events this looks quite strange. Getnewinventory getoldinventory?

File: src/main/java/org/spongepowered/api/event/block/data/FurnaceConsumeFuelEvent.java [hunk 1]
```diff
@@ -35,20 +37,24 @@
...
      *
      * <p>A {@link Furnace} uses {@link ItemStack}s to fuel itself, and after the fuel is
      * spent, the item is burned.</p>
      *
+     * <p>In Vanilla, the returned {@link Inventory} can be queried for {@link FuelSlot}.</p>
+     *
      * @return The burned item
      */
-    ItemStack getBurnedItem();
+    Inventory getBurnedItem();
```
Comment (me4502) [inline src/main/java/org/spongepowered/api/event/block/data/FurnaceConsumeFuelEvent.java]: I think these methods should probably be renamed.

**Comment by modwizcode** [comment]
This needs to be done but these aren't the correct ways to do this. These cannot keep the current names. 

Some of the events work better with itemstacks also, an equipment changed event should have the item(s) that changed, not an inventory, it's nice to have the inventory too, but the important part is the items themselves, which are better represented as a List<ItemStack> or a single ItemStack. Smelting also, having the single slots present in the event is for convience and having two inventories for a single event that are referring to two items doesn't make sense either.

This is not a thoughtful look at the problem of adding inventory integration into the existing parts of the API. 

I recommend you start over, because this isn't a good starting point.

**Comment by Zidane** [comment]
@Mumfrey 

Seeking comments from you on this.

**Comment by Mumfrey** [comment]
@Zidane I agree with @modwizcode, this is sound in principle but not in execution. Certain things make sense to return either an `ItemStack` or (maybe preferably) a particular type of `Slot` (which is an `Inventory` at the end of the day) and anything which just generically *has* an `Inventory` getter should be a `Carrier` rather than having a specialised `getWhateverInventory` accessor.

**Comment by Zidane** [comment]
@Mumfrey knows better than I do. Closing this until a better solution is presented.

**Pull request closed without merging**